packages/functional-tests/src/tests/grid-test.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import * as MRE from '@microsoft/mixed-reality-extension-sdk';
+
+import { Test } from '../test';
+
+export default class GridTest extends Test {
+	public expectedResultDescription = "Click balls to realign grid";
+	private assets: MRE.AssetContainer;
+
+	public async run(root: MRE.Actor): Promise<boolean> {
+		this.assets = new MRE.AssetContainer(this.app.context);
+		const box = this.assets.createBoxMesh('box', 0.24, 0.24, 0.24);
+		const ball = this.assets.createSphereMesh('ball', 0.15);
+
+		const anchor = MRE.Actor.Create(this.app.context, { actor: {
+			name: "anchor",
+			parentId: root.id,
+			transform: { local: { position: new MRE.Vector3(0, 1, -1)}},
+			appearance: { meshId: box.id }
+		}});
+
+		// create button grid
+		const buttonGrid = new MRE.PlanarGridLayout(anchor);
+		for (let i = 0; i < 9; i++) {
+			const alignment = Object.values(MRE.BoxAlignment)[i];
+			const button = MRE.Actor.Create(this.app.context, { actor: {
+				name: alignment + "-button",
+				parentId: anchor.id,
+				appearance: { meshId: ball.id },
+				collider: { geometry: { shape: MRE.ColliderType.Auto } }
+			}});
+			button.setBehavior(MRE.ButtonBehavior).onClick(() => {
+				buttonGrid.gridAlignment = alignment;
+				buttonGrid.applyLayout();
+			});
+			buttonGrid.addCell({
+				row: Math.floor(i / 3),
+				column: i % 3,
+				width: 0.3,
+				height: 0.3,
+				contents: button
+			});
+
+			const label = MRE.Actor.Create(this.app.context, { actor: {
+				name: "label",
+				parentId: anchor.id,
+				text: {
+					contents: GridTest.ShortName(alignment),
+					height: 0.05,
+					anchor: GridTest.BoxToTextAlignment(alignment),
+					color: MRE.Color3.Teal()
+				}
+			}});
+			buttonGrid.addCell({
+				row: Math.floor(i / 3),
+				column: i % 3,
+				width: 0.3,
+				height: 0.3,
+				alignment,
+				contents: label
+			});
+		}
+		buttonGrid.applyLayout();
```

Comment:
>applyLayout [](start = 13, length = 11)

How about making this animate?

Reply:
Why not?

---
In reply to: [385873835](https://github.com/microsoft/mixed-reality-extension-sdk/pull/492#discussion_r385873835) [](ancestors = 385873835)